packages/babylon/src/parser/statement.js

```diff
@@ -1368,6 +1368,10 @@ export default class StatementParser extends ExpressionParser {
       return this.parseFunction(expr, true, false, true, true);
     } else if (this.match(tt._class)) {
       return this.parseClass(expr, true, true);
+    } else if (this.match(tt.at)) {
+      this.parseDecorators(false);
+      // startNode has changed
+      return this.parseClass(this.startNode(), true, false, false, true);
```

Comment:
Q: Why do we need to `this.startNode()` instead of using `expr`?

Reply:
i think expr will still point to the node which is of decorators. After parsing decorators, current node will point to class Node which is needed to be parsed.

Reply:
The `parseDecorator` function (called by `parseDecorators`) creates its own node, so `expr` won't be used. If you have  tried and using `expr` breaks something leave it as it is now, otherwise don't create an unnecessary node :wink:

Reply:
Yes, it creates its own node which according to me, is required.
For example
```js
export default @dec class x{}
```
In the code above `expr` will point to the node having `start` at 15 (when expr is assigned then current node is still `tt.at`) so when `class` is parsed without creating an extra node, it's current `start` will be pointed by 15 which acc. to me should be 20. that's why we need to create extra Node, which will point to current location of class i.e. 20. I've seen that class declaration/expression nodes start at the starting of class keyword. I maybe wrong though.. :grimacing:

Reply:
@nicolo-ribaudo, i cross-checked by using expr instead of creating a new node, and tested it for following input:-
```js
export default @dec class A {}
```
, i got this output:- 
```js
[ Node {
    type: 'ExportDefaultDeclaration',
    start: 0,
    end: 30,
    loc: SourceLocation { start: [Object], end: [Object] },
    declaration:
     Node {
       type: 'ClassDeclaration',
       start: 15,
       end: 30,
       loc: [Object],
       decorators: [Array],
       id: [Object],
       superClass: null,
       body: [Object] } } ]
```
However the ouptut must be :- 
```js
[ Node {
    type: 'ExportDefaultDeclaration',
    start: 0,
    end: 30,
    loc: SourceLocation { start: [Object], end: [Object] },
    declaration:
     Node {
       type: 'ClassDeclaration',
       start: 20,
       end: 30,
       loc: [Object],
       decorators: [Array],
       id: [Object],
       superClass: null,
       body: [Object] } } ]
```
I think we need to create new node. However, suggestions/corrections are appreciated :slightly_smiling_face: